plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -955,8 +956,27 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
                             .map(dataFile -> dataFile.path().toString())
                             .collect(toImmutableSet()));
         }
+        // The idea is that after optimize there should not be any delete files used anymore as
+        // the new data files have all these deletes already applied. This is possible
+        // due to the fact that we prevent skipping any data files in
+        // io.trino.plugin.iceberg.IcebergSplitSource.getNextBatch when delete files are present.
+        Table tableForScanning = catalog.loadTable(session, executeHandle.getSchemaTableName());
+        long currentSequenceNumber = icebergTable.currentSnapshot().sequenceNumber();
+        Optional<Snapshot> lastSnapshot = stream(tableForScanning.snapshots())
+                .filter(snapshot -> snapshot.sequenceNumber() == currentSequenceNumber)
+                .findFirst();
```

Comment:
is it equivalrnt to `table.snapshot(long)`?

Reply:
Yes, I should do it differently, originally I thought I needed currentSequence - 1 and then didn't change it properly. But it all doesn't matter as it is incorrect because of the thing you mentioned below.